R/drive_mkdir.R

```diff
@@ -1,35 +1,68 @@
 #' Create a folder on Google Drive.
 #'
-#' @param name Character. The name of the folder you would like to create.
 #' @template path
+#' @param name Character. The name of the folder you would like to create, if
+#'   not already specified in `path`.
 #' @template verbose
 #'
 #' @template dribble-return
+#'
+#' @examples
+#' \dontrun{
+#' ## Create folder named "def" in existing folder "abc".
+#' drive_mkdir("abc/def")
+#'
+#' ## This will also create a folder named "def" in folder "abc".
+#' drive_mkdir(path = "abc", name = "def")
+#'
+#' ## Another way to create a folder named "def" in folder "abc",
+#' ## this time with parent folder stored in a dribble.
+#' abc <- as_dribble("abc")
+#' drive_mkdir(path = abc, name = "def")
+#'
+#' ## Yet another way to do this,
+#' ## this time with parent folder provide via id.
+#' drive_mkdir(path = as_id(abc$id), name = "def")
+#'}
 #' @export
-drive_mkdir <- function(name = NULL, path = NULL, verbose = TRUE) {
-  parent <- NULL
+drive_mkdir <- function(path = NULL, name = NULL, verbose = TRUE) {
+  if (!is.null(name)) {
+    stopifnot(is_path(name), length(name) == 1)
+  }
 
-  if (!is.null(path)) {
-    if (inherits(path, "character")) {
+  if (is_path(path)) {
+    if (is.null(name)) {
+      path <- strip_slash(path)
+    } else {
       path <- append_slash(path)
     }
```

Comment:
Contradicting what I said last time we met: in the end, it made sense to consult the presence/absence of explicit `name` when deciding how to interpret `path`. I was wrong 😬.